### DISCUSSION TITLE
Trying to get Micropython on the M5Stamp-S3 (ESP32-S3)

### DISCUSSION THREAD

**Discussion by echo-lalia** [micropython/micropython#13545] [ESP32] [answered]
I recently got an M5Stack Cardputer, which uses the "M5Stamp-S3" board to control it. That board is based on the ESP32-S3.

It's a pretty neat little device that I find really cute. I'd love to develop some stuff on the thing, but I already have a lot of experience with Micropython on the Pi Pico, and I would definitely prefer to continue to use Micropython than have to learn to use something new for this. 

<br>
<br>

I've tried flashing the generic ESP32-S3 firmware, as well as some other firmwares made for other ESP32-S3 boards, but no luck. When I try, I can't get access to it in Thonny, and the board just stops responding over USB until I put it into bootloader mode and reflash it.

<br>
<br>

I see that there are already some official ESP32-S3 ports available, and I also see that some other M5Stack products have official ports. I've tried reading the installation instructions for those other ports, and did some searching about the differences between the devices, and how new boards get supported in Micropython, but I'm not really grasping why this thing isn't just working with the generic ESP32-S3 Micropython port. I've also tried using some different settings/utilities for flashing, and being sure to erase the flash between attempts, but no luck. 

Anyways, I'm curious if anyone else has tried messing with this, has some insight to offer as to how I could figure out what the problem is, or happens to know what the issue here is already :)

**Comment by jimmo** [comment]
Can you post the output of running esptool.py to flash the firmware?

**Reply by echo-lalia** [reply]
Sure! I can do that:

```
:~$ esptool.py --baud 460800 write_flash -z 0x0 /home/echo/Downloads/ESP32_GENERIC_S3-20240105-v1.22.1.bin
esptool.py v4.7.0
Found 1 serial ports
Serial port /dev/ttyACM0
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (QFN56) (revision v0.2)
Features: WiFi, BLE, Embedded Flash 8MB (GD)
Crystal is 40MHz
MAC: 34:b7:da:56:3f:08
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x0018dfff...
Compressed 1628240 bytes to 1066973...
Wrote 1628240 bytes (1066973 compressed) at 0x00000000 in 9.8 seconds (effective 1323.1 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

**Reply by jimmo** [reply]
And you did an erase_flash first?

**Reply by echo-lalia** [reply]
Yup: 
```
:~$ esptool.py --baud 460800 erase_flash
esptool.py v4.7.0
Found 1 serial ports
Serial port /dev/ttyACM0
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (QFN56) (revision v0.2)
Features: WiFi, BLE, Embedded Flash 8MB (GD)
Crystal is 40MHz
MAC: 34:b7:da:56:3f:08
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Erasing flash (this may take a while)...
Chip erase completed successfully in 2.7s
Hard resetting via RTS pin...
```

**Comment by shariltumin** [comment]
Have you tried a hardware reset by briefly connecting the EN pin to ground after flashing the firmware?

**Reply by echo-lalia** [reply, marked as answer]
Darn, I spent like the entire day yesterday on this, and just realized I was being silly 😂

I just flashed the generic firmware of the recent build, and it actually seems to be working. 

It changes the USB interface and also isn't detected automatically anymore after being flashed, so I couldn't find it and thought it was just broken. Turns out I just had to find the correct port to connect to it, and it's working now!

**Reply by lc-at** [reply]
Hi @echo-lalia, how did you manage to get MicroPython to work on the Cardputer? After flashing the generic firmware, I can't seem to have the serial REPL working. It is always detected as Espressif USB JTAG/serial debug unit.

**Comment by echo-lalia** [comment]
@ttycelery 

I've actually made step-by-step instructions for this. You can find it [here](https://github.com/echo-lalia/Cardputer-MicroHydra) in the MicroHydra readme. It includes steps for installing MicroPython with Thonny (and then for installing MicroHydra but you can ignore that part).

**Reply by lc-at** [reply]
Thanks!
